src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -537,6 +538,23 @@ export class KycService {
     return KycStepMapper.toStepBase(kycStep);
   }
 
+  async updatePaymentData(kycHash: string, stepId: number, data: PaymentDataDto): Promise<KycStepBase> {
+    const user = await this.getUser(kycHash);
+    const kycStep = user.getPendingStepOrThrow(stepId);
+
+    if (data.contractAccepted) {
+      await this.userDataService.updateUserDataInternal(user, { paymentLinksAllowed: true });
+      // TODO: send mail with text
+    }
+
+    return this.updateKycStepAndLog(
+      kycStep,
+      user,
+      data,
+      data.contractAccepted ? KycStepStatus.COMPLETED : KycStepStatus.FAILED,
```

Comment:
I think ManualReview could make more sense for Monitoring etc. to notice the decline and maybe call the customer if needed 🤔